src/library/basesqltablemodel.cpp

```diff
@@ -293,7 +324,7 @@ void BaseSqlTableModel::select() {
     // should not disturb that if we are only removing tracks.
     qStableSort(rowInfo.begin(), rowInfo.end());
 
-    m_trackIdToRows.clear();
+    TrackId2Rows trackIdToRows;
```

Comment:
1st select: Initial select after startup
2nd select: Switch sort order of Date Added column

```
master
------
1st select: Debug [Main]: LibraryTableModel(0x363dfc0) select() took 2568 ms 217149 
2nd select: Debug [Main]: LibraryTableModel(0x363dfc0) select() took 2509 ms 217149 

basesqltablemodel_members
-------------------------
1st select: Debug [Main]: LibraryTableModel(0x363dec0) select() took 2536 ms 217149 
2nd select: Debug [Main]: LibraryTableModel(0x363dec0) select() took 2450 ms 217149 
```

All timings vary around 2500 ms and there is no clear winner.